### PR TITLE
Feature/add filter polygons

### PR DESCRIPTION
**Pull request by chahank**
Add mini-feature to filter out polygons.

### CODE REVIEW

**Comment by chahank** [comment]
No idea why the test fails. It says gpd.testing does not exist, yet it is clearly in geopandas https://geopandas.org/en/stable/docs/reference/testing.html

File: src/osm_flex/simplify.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+OSM-flex is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+-----
+simplification functions
+"""
+
+def remove_small_polygons(gdf, min_area):
+    """Remove (multi-)polygons of area smaller than min_area
+    Points and lines are untouched.
```
Comment (Evelyn-M) [inline src/osm_flex/simplify.py:16]: would be nice to have the option of a metres=True kwarg (similar to the CLIMADA interpolation funcs), where min_area can hence be specified in m2, irrespective of the original projection of the gdf.
Requires a check & potential re-projection of the gdf under the hood.

Comment (chahank) [reply src/osm_flex/simplify.py:16]: I think that would be nice for a future extension. Shall we already merge this first version?

**Comment by chahank** [comment]
Good fixes! I think we can merge.

File: src/osm_flex/simplify.py (hunk 1 above)
Comment (Evelyn-M) [reply src/osm_flex/simplify.py:16]: hmm, I'd add this right away maybe, so we don't have to change the signature of the function later on

**Comment by Evelyn-M** [comment]
I'm not so sure anymore whether we should really have any `simplification` methods in osm-flex at all, especially given that there is a lot of code out there and currently being developed that will deal with these spatial operations in a more rigorous way (e.g. Sahand's work). Opinions? @ElcoK @chahank ?

**Comment by chahank** [comment]
I would be in favor of still having the really minimalistic ones in here, i.e. those that are not specific to applications (if there are any). But I am also not really aware of other options, so that might also be the reason.

**Comment by Evelyn-M** [comment]
Ok. Then I now added a few more "simple simplification" methods, such that this module is not a stub:
- dropping exact geometry duplicates
- dropping points that are contained in polygons
- dropping polygons that are contained in polygons

The methods still need tests and a small tutorial in the docs.

**Comment by Evelyn-M** [comment]
I added tests to the `simplify` module, and also a small tutorial. Everything is passing, hence I'd merge- @chahank, @ElcoK ?

**Pull request merged**